sub-packages/bionemo-size-aware-batching/tests/bionemo/size_aware_batching/test_sampler.py

```diff
@@ -486,6 +486,8 @@ def test_iter_bucket_batch_sampler(sample_data):
 
 
 def test_iter_bucket_batch_sampler_with_shuffle(sample_data):
+    torch.set_default_device("cuda")
+
```

Comment:
this is going to impact other tests, right?

Reply:
My preference would be to make a context manager that has the flow:
1. get current default device (maybe "cpu" if unset)
2. set to user requested new default device
3. on exit restore default device regardless of error state etc in finally block

```python
from contextlib import contextmanager
@contextmanager
def default_device_ctx(new_device):
    try:
       original_device = torch.... # whatever this is
       torch.set_default_device(new_device)
       yield
    finally:
        torch.set_defulat_device(original_device)
```

Reply:
Also why do you need this? maybe there are torch.tensor... inside of `BucketBatchSampler` that are not placed on the right device in the context that sizes and/or bucket_boundaries are tensors on some device? Seems like a hack to get stuff working potentially when there are other problems underneath.

Reply:
As you suggested in the slack, we remove this part to avoid any device control, and refactor the check to not do numerical comparison, but check some basic logics.